packages/google_maps_flutter/google_maps_flutter_web/example/integration_test/overlay_test.dart

```diff
@@ -0,0 +1,118 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:convert';
+import 'dart:html' as html;
+
+import 'package:flutter_test/flutter_test.dart';
+import 'package:google_maps/google_maps.dart' as gmaps;
+import 'package:google_maps_flutter_platform_interface/google_maps_flutter_platform_interface.dart';
+import 'package:google_maps_flutter_web/google_maps_flutter_web.dart';
+import 'package:integration_test/integration_test.dart';
+
+import 'resources/tile16_base64.dart';
+
+class NoTileProvider implements TileProvider {
+  const NoTileProvider();
+
+  @override
+  Future<Tile> getTile(int x, int y, int? zoom) async => TileProvider.noTile;
+}
+
+class TestTileProvider implements TileProvider {
+  const TestTileProvider();
+
+  @override
+  Future<Tile> getTile(int x, int y, int? zoom) async =>
+      Tile(16, 16, const Base64Decoder().convert(tile16Base64));
+}
+
+/// Test Overlays
+void main() {
+  IntegrationTestWidgetsFlutterBinding.ensureInitialized();
+
+  group('TileOverlayController', () {
+    const TileOverlayId id = TileOverlayId('');
+    late TileOverlayController controller;
+
+    setUp(() {
+      controller = TileOverlayController();
+    });
+
+    testWidgets('minimal initialization', (WidgetTester tester) async {
+      controller.update(const TileOverlay(tileOverlayId: id));
+
+      final gmaps.Size size = controller.gmMapType.tileSize!;
+      expect(size.width, TileOverlayController.logicalTileSize);
+      expect(size.height, TileOverlayController.logicalTileSize);
+      expect(controller.gmMapType.getTile!(gmaps.Point(0, 0), 0, html.document),
+          null);
+    });
+
+    testWidgets('produces image tiles', (WidgetTester tester) async {
+      controller.update(const TileOverlay(
+        tileOverlayId: id,
+        tileProvider: TestTileProvider(),
+      ));
+
+      final html.ImageElement img =
+          controller.gmMapType.getTile!(gmaps.Point(0, 0), 0, html.document)!
+              as html.ImageElement;
+      expect(img.naturalWidth, 0);
+      expect(img.naturalHeight, 0);
+
```

Comment:
We used to expect that `img.hidden` was `true` before load, and `false` afterwards. Is this really required? In my testing I didn't see any difference and removed the attribute. @AsturaPhoenix / @elitree do you remember why was the attribute added?

Reply:
I think it was to prevent the image decoding from blocking the main thread, but I can't find a version without hidden + decode so I'm not sure I ever ran a comparison. In any case we still have the bottleneck of populating the image data on the Dart/Flutter side, which has to happen on the main thread on web today.

I do think there's a difference in the case of bad image data though; if `Tile.data` is not a valid image, I think adding the `img` straight and setting the `src` will eventually make it display a browser-dependent broken image whereas hiding and decoding wouldn't, but I don't think that was the primary motivation.